diff --git a/core/src/main/java/org/bitcoin/Secp256k1Context.java b/core/src/main/java/org/bitcoin/Secp256k1Context.java
index c1d6484..867e36f 100644
--- a/core/src/main/java/org/bitcoin/Secp256k1Context.java
+++ b/core/src/main/java/org/bitcoin/Secp256k1Context.java
@@ -33,7 +33,7 @@ public class Secp256k1Context {
         boolean isEnabled = true;
         long contextRef = -1;
         try {
-            System.loadLibrary("secp256k1");
+            System.loadLibrary("wallycore");
             contextRef = secp256k1_init_context();
         } catch (UnsatisfiedLinkError e) {
             log.info(e.toString());
